client-runtime/src/main/java/com/microsoft/rest/RestProxy.java

```diff
@@ -136,25 +136,38 @@ private HttpRequest createHttpRequest(SwaggerMethodParser methodParser, Object[]
         final Object bodyContentObject = methodParser.body(args);
         if (bodyContentObject != null) {
             String contentType = methodParser.bodyContentType();
+            if (contentType == null || contentType.isEmpty()) {
+                contentType = request.headers().value("Content-Type");
+            }
+            if (contentType == null || contentType.isEmpty()) {
+                if (bodyContentObject instanceof byte[] || bodyContentObject instanceof String) {
+                    contentType = ContentType.APPLICATION_OCTET_STREAM;
+                }
+                else {
+                    contentType = ContentType.APPLICATION_JSON;
```

Comment:
Down the line we're going to hit funny stuff related to XML support in blob storage API (as far as I understand). C# people recently added this onto their runtime. Probably no need to address it in this PR but something to be aware of. May even be necessary to set a different serializer adapter in gen code upon detecting some attribute in the service.